### PR TITLE
Add MacCamy-Fuchs and Cd correction options

### PR DESCRIPTION
**Pull request by mattcapa**
This PR adds optional corrections as follows:
- MacCamy-Fuchs correction for large cylinders
- Cd correction for large cylinders

### CODE REVIEW

**Comment by tridelat** [comment]
Adding some details on the workflow of this PR for the record.
Since the MacCamy-Fuchs correction tables depend on wave peak period and need to be generated only once, only one correction table object is declared and instantiated with `extern` (essentially acting as a singleton), and is accessible from the the seahowl::hydro namespace.
When environmental conditions are generated, the wave peak period of the waves (if any) is associated to the correction table. The added mass / drag coefficients are then interpolated by calling the table with the Morison node diameters iff diameters are large enough and fluid density is above a certain threshold (set to 500.0 here, meaning that no correction is made if the Morison node is in the air phase).

File: include/seahowl/aero/tower_aero.h [hunk 1]
```diff
@@ -37,6 +37,10 @@ class TowerAero : public ComponentFluid {
     std::vector<hydro::MorisonElement> elements;
     /** @brief Loads at center of tower elements. */
     std::vector<Vector3d> loads;
+    /** @brief MacCamy and Fuchs Correction for large cylinders, Flag. */
+    bool use_MacCamyFuchs_correction = true;
+    /** @brief Cd Correction for large cylinders, Flag. */
+    bool use_Cd_correction;
```
Comment (tridelat) [inline include/seahowl/aero/tower_aero.h]: Initialize variable for use_Cd_correction (true/false). Maybe both variables should be initialized at false to force user to explicitly activate it in .json/python?

File: include/seahowl/env/wave_models.h [hunk 2]
```diff
@@ -19,6 +19,8 @@ class WaveModel : public FluidModel {
     Vector3d surface_normal{0.0, 0.0, 1.0};
     /** @brief Water depth. */
     double water_depth = 0.0;
+    /** @brief MacCamy-Fuchs correction flag. */
+    bool MacCamyFuchsflag = true;
```
Comment (tridelat) [inline include/seahowl/env/wave_models.h]: variable not used, can be removed (the flag is on the tower)

File: src/hydro/mooring_hydro.cpp [hunk 3]
```diff
@@ -10,7 +10,9 @@ using namespace seahowl;
 using namespace seahowl::hydro;
 using seahowl::env::FluidModel;
 
-MooringHydro::MooringHydro() {}
+MooringHydro::MooringHydro() {
+    coefficients.use_MacCamyFuchs_correction = false;
```
Comment (tridelat) [inline src/hydro/mooring_hydro.cpp:14]: add explicitly coefficients.use_Cd_correction = false;

**Review by tridelat — CHANGES_REQUESTED** [review]

File: include/seahowl/aero/tower_aero.h (hunk 1 above)
Comment (mattcapa) [reply include/seahowl/aero/tower_aero.h]: I propose to initialize both with true.. I think those corrections should the norm. 
A user aiming at testing a specific Cd or Cm, it should force the flags to false.

File: include/seahowl/env/wave_models.h (hunk 2 above)
Comment (mattcapa) [reply include/seahowl/env/wave_models.h]: very good check ;)

File: src/hydro/mooring_hydro.cpp (hunk 3 above)
Comment (mattcapa) [reply src/hydro/mooring_hydro.cpp:14]: ok

File: include/seahowl/aero/tower_aero.h (hunk 1 above)
Comment (tridelat) [reply include/seahowl/aero/tower_aero.h]: Okay, let's keep it this way

**Review by tridelat — APPROVED** [review]
Good to go, thanks!

**Pull request merged**